adoc/chapters/cuda_backend.adoc

```diff
@@ -0,0 +1,496 @@
+| [code]#sampled_image_accessor<T, 2, Mode, image_target::device>#   | sycl::cuda::texture<T, 2> | sycl::cuda::texture<T, 1> | Convert a SYCL `accessor` to the `sycl::cuda::texture` interoperability type with the same type and dimensions.
+| [code]#sampled_image_accessor<T, 3, Mode, image_target::device>#   | sycl::cuda::texture<T, 3> | sycl::cuda::texture<T, 1> | Convert a SYCL `accessor` to the `sycl::cuda::texture` interoperability type with the same type and dimensions.
+| [code]#unsampled_image_accessor<T, 1, Mode, image_target::device># | sycl::cuda::surface<T, 1> | sycl::cuda::surface<T, 1> | Convert a SYCL `accessor` to the `sycl::cuda::surface` interoperability type with the same type and dimensions.
+| [code]#unsampled_image_accessor<T, 2, Mode, image_target::device># | sycl::cuda::surface<T, 2> | sycl::cuda::surface<T, 1> | Convert a SYCL `accessor` to the `sycl::cuda::surface` interoperability type with the same type and dimensions.
+| [code]#unsampled_image_accessor<T, 3, Mode, image_target::device># | sycl::cuda::surface<T, 3> | sycl::cuda::surface<T, 1> | Convert a SYCL `accessor` to the `sycl::cuda::surface` interoperability type with the same type and dimensions.
+| [code]#stream#                                                     | signed char * | signed char * | Convert a SYCL `accessor` to an undecorated raw signed char pointer.
+|====
+
+
+[[sec:non_core_features_and_extensions]]
+== Non-core features and extensions
+
+Additional CUDA features are available depending upon the devices compute capability.
+SYCL can support these optional CUDA features with extensions.
+Unlike OpenCL, CUDA needs to know if the extension is available at compile time. 
+As a result there are no valid CUDA extensions which can be passed to `has_extension`.
+
+As the extension must be known at runtime CUDA extensions are best implemented 
+using feature test macros. The test macro format is 
+SYCL_EXT_<vendorstring>_<featurename>. For CUDA extensions this format translates 
+to SYCL_EXT_NVIDIA_<featurename>. Similarly, the format for the naming of extension 
+classes and enumerations should be ext_<vendorstring>_<featurename>. Which in this context
+becomes ext_NVIDIA_<featurename>. Given the necessity to know the extension at 
+compile-time, the usage of extension macros should be the primary method of determining 
+if the extension is available in the SYCL implementation not. 
+A list of non-core CUDA features which have SYCL support is below.
+Non-core CUDA features for require a compute capability of greater than 5.
+
+TODO: The table below shows a proposal for SYCL supported CUDA extensions.
+The table should be developed with other members of the SYCL community.
+
+[[table.extensionsupport]]
+.SYCL support for CUDA 11.3 extensions
```

Comment:
I think the idea of adding a column for feature test macros is a good idea. We want to discourage the use of sycl::aspects to determine the availability of extensions, since the extensions need to be configured at compile time rather than runtime. Therefore we will replace the SYCL Aspect column with a feature test macro column.

We can also remove the Core SYCL API column

Reply:
sycl::aspects describing extensions are useful to check if a particular device has a capability at runtime. Do you think any mention of such aspects is outside the scope of this document?

Reply:
We've updated the table to reflect the associated SYCL aspect and feature macro for each extension, the table is currently empty, but serves as a placeholder for us to populate in the future.